packages/context/src/binding-inspector.ts

```diff
@@ -272,11 +287,13 @@ function buildBindingKey(
   cls: Constructor<unknown>,
   options: BindingFromClassOptions = {},
 ) {
+  if (options.key) return options.key;
```

Comment:
Small optimization - we don't need to call `buildingTemplateFor` and create a new `Binding` when `options.key` is set. Return as early as possible in that case.